sharktank/sharktank/layers/testing.py

```diff
@@ -43,3 +53,106 @@ def make_llama_attention_block_theta(
             ),
         }
     )
+
+
+class CausalLMIreeModel(BaseCausalLMModel, CausalLMModelABC):
```

Comment:
Dependent on the double inheritance PR you have. This should have a cleaner design rather than extend with double inheritance.

Reply:
I removed `CausalLMModelABC`.